lib/dependencies/ContextDependencyTemplateAsRequireCall.js

```diff
@@ -2,38 +2,41 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function ContextDependencyTemplateAsRequireCall() {}
-module.exports = ContextDependencyTemplateAsRequireCall;
+"use strict";
 
-ContextDependencyTemplateAsRequireCall.prototype.apply = function(dep, source, outputOptions, requestShortener) {
-	var comment = "";
-	if(outputOptions.pathinfo) comment = "/*! " + requestShortener.shorten(dep.request) + " */ ";
-	var containsDeps = dep.module && (dep.module.dependencies && dep.module.dependencies.length > 0);
-	var isAsync = dep.module && dep.module.async;
-	if(dep.module && (isAsync || containsDeps)) {
-		if(dep.valueRange) {
-			if(Array.isArray(dep.replaces)) {
-				for(var i = 0; i < dep.replaces.length; i++) {
-					var rep = dep.replaces[i];
-					source.replace(rep.range[0], rep.range[1] - 1, rep.value)
+class ContextDependencyTemplateAsRequireCall {
+
+	apply(dep, source, outputOptions, requestShortener) {
+		let comment = "";
+		if(outputOptions.pathinfo) comment = "/*! " + requestShortener.shorten(dep.request) + " */ ";
+		const containsDeps = dep.module && (dep.module.dependencies && dep.module.dependencies.length > 0);
+		const isAsync = dep.module && dep.module.async;
+		if(dep.module && (isAsync || containsDeps)) {
+			if(dep.valueRange) {
+				if(Array.isArray(dep.replaces)) {
+					for(let i = 0; i < dep.replaces.length; i++) {
```

Comment:
Hm... 🤔

```js
if (dep.replaces && dep.replaces[Symbol.iterator]) {
  for (const rep of dep.replaces) {
    source.replace(rep.range[0], rep.range[1] - 1, rep.value)
  }
}
```

Checking for `Symbol.iterator` and using `for of` instead of `Array.isArray` + `for` makes this more generic and support arbitrary iterables (including arrays and iterator instances), but I wonder if @sokra is ok with the overhead.

Even if the check stays `Array.isArray` instead of the generic `Symbol.iterator` check, this probably should be either a `for of` or `dep.replaces.forEach` call.